scripts/sandbox.py

```
@@ -57,15 +58,34 @@ def main() -> int:
     return 0
 
 
+def extract_build_args(remaining: list[str]) -> dict[str, str]:
+    """Extract --build-arg KEY=VALUE pairs from the command line using argparse."""
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument("--build-arg", action="append", default=[])
+    known, _ = parser.parse_known_args(remaining)
+    build_args = {}
+    for arg in known.build_arg:
+        if "=" not in arg:
+            raise ValueError(f"--build-arg must be in KEY=VALUE format, got: {arg!r}")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)